src/peer_manager.rs

```diff
@@ -560,7 +542,17 @@ impl PeerManager {
                 if tunnel {
                     Err(RoutingError::CandidateIsTunnelling)
                 } else {
-                    Ok(Some((candidate.seed.clone(), candidate.target_size)))
+                    let (target_size, seed) = if difficulty == 0 {
+                        (10, vec![5u8; 4])
+                    } else {
+                        let evaluators = cmp::max(self.routing_table.min_section_size(),
+                                                  self.routing_table.our_section().len());
+                        // TODO: the algorithm need to be measured and refactored against the real network
+                        (2 * 1024 * 1024 / evaluators,
+                         rand::thread_rng().gen_iter().take(10).collect())
+                    };
+                    candidate.resource_proof = Some((target_size, difficulty, seed.clone()));
```

Comment:
>But that problem is there with or without this PR, isn't it?
in this scenario, the current code calculates target_size before send_out get_node_name_response. which reflects the num of connections the joining node needs to establish.

>have to reply successfully to all the challenges, instead of just a quorum
the current code also requires that, but with a larger target_size, so the new approach is easier compared with the current.

Anyhow, this is really an edge case and the network shall be ok to stay with it and using the new approach.